recipes/users.rb

```diff
@@ -48,10 +48,15 @@ def chef_solo_search_installed?
       not_if { ::File.exist?("#{node['openvpn']['key_dir']}/#{u['id']}.crt") }
     end
 
+    node.set['openvpn']['client_config']['cert'] = "#{u['id']}.crt"
+    node.set['openvpn']['client_config']['key']  = "#{u['id']}.key"
```

Comment:
Expanded the code here to see where `u['id']` was coming from and I'm thinking this should be changed as you're doing a global `node.set` inside a dynamic list of users.  IIRC, this will cause `node[openvpn][client_config][cert]` to be set to the last user processed and all templates will end up using that value.  Probably want to set the values in a local hash, then merge that into `node[openvpn][client_config]` when the template is evaluated, to avoid impacting global settings.

Example:
```ruby
local_config = {
  'cert' => "#{u['id']}.crt",
  'key' => "#{u['id']}.key"
}
local_config.merge!(node['openvpn']['client_config'])

template ... do
  variables(
    config: local_config
  )
end
```

If someone sets their own custom cert/key, according to [Ruby Hash .merge](https://ruby-doc.org/core-2.3.1/Hash.html#method-i-merge-21), it will override the local_config value, which seems correct